### PR TITLE
Add WorkQueue debugging for ACDC - NEVER merge

### PR DESCRIPTION
**Pull request by amaltaro**
To debug #8220
@ticoann this is the patch I suggest we apply to the agents for a few weeks until we can have a better insight of the problem. Can you have a quick look? I'll also test it in testbed first, just to make sure it isn't broken.

### CODE REVIEW

File: src/python/WMCore/WorkQueue/WorkQueueBackend.py [hunk 1]
```diff
@@ -416,7 +416,7 @@ def availableWork(self, thresholds, siteJobCounts, teams=None, wfs=None,
                 siteJobCounts[possibleSite][prio] = siteJobCounts[possibleSite].setdefault(prio, 0) + \
                                                     element['Jobs'] * element.get('blowupFactor', 1.0)
             else:
-                self.logger.info("No available resources for %s with doc id %s", element['RequestName'], element.id)
+                self.logger.debug("No available resources for %s with doc id %s", element['RequestName'], element.id)
```
Comment (ticoann) [inline src/python/WMCore/WorkQueue/WorkQueueBackend.py]: Alan, shouldn't this be separate patch and need to be applied not for the testing?

File: src/python/WMCore/WMBS/File.py [hunk 2]
```diff
@@ -615,6 +615,8 @@ def addFilesToWMBSInBulk(filesetId, workflowName, files, isDBS = True,
                                 conn = conn,
                                 transaction = transaction)
     if len(runLumiBinds) > 0:
+        logging.info("AMR about to add run/lumi for %i files for fileset %i",
+                     len(runLumiBinds), filesetId)
```
Comment (ticoann) [inline src/python/WMCore/WMBS/File.py]: what is the structure of runLumiBinds? can we can the all the information we need?

File: src/python/WMCore/WorkQueue/WorkQueue.py [hunk 3]
```diff
@@ -412,6 +412,16 @@ def _getDBSBlock(self, match, wmspec):
...
                                            splitedBlockName['NumOfFiles'])
+            ### FIXME temporary debugging  ###
+            self.logger.info("AMR processing %d non-unique files for fileset %s",
+                             len(fileLists), acdcInfo['fileset'])
+            lumisPerRun = {}
+            for entry in fileLists:
+                for run in entry['runs']:
+                    lumisPerRun.setdefault(run.run, 0)
+                    lumisPerRun[run.run] += len(run.lumis)
+            self.logger.info("AMR run and # of lumis to be recovered: %s", lumisPerRun)
```
Comment (ticoann) [inline src/python/WMCore/WorkQueue/WorkQueue.py]: do we need information for all the run lumis not by file?

File: src/python/WMCore/WorkQueue/WorkQueue.py [hunk 4]
```diff
@@ -412,6 +412,16 @@ def _getDBSBlock(self, match, wmspec):
                                            acdcInfo['fileset'],
                                            splitedBlockName['Offset'],
                                            splitedBlockName['NumOfFiles'])
+            ### FIXME temporary debugging  ###
+            self.logger.info("AMR processing %d non-unique files for fileset %s",
+                             len(fileLists), acdcInfo['fileset'])
```
Comment (ticoann) [inline src/python/WMCore/WorkQueue/WorkQueue.py]: What kind of information we are tracking?

**Comment by ticoann** [comment]
Hi Alan, what you observed was the difference between acdc record and wmbs record, correct?
I am not completely sure about what we can find out with debugging message.

File: src/python/WMCore/WorkQueue/WorkQueueBackend.py (hunk 1 above)
Comment (amaltaro) [reply src/python/WMCore/WorkQueue/WorkQueueBackend.py]: agreed! I'm making a new patch to apply with this one

File: src/python/WMCore/WMBS/File.py (hunk 2 above)
Comment (amaltaro) [reply src/python/WMCore/WMBS/File.py]: it's a list of dictionaries, with 2 keys/values, lfn and Run object.
Ideally we should print everything, the problem is that there are files (like MCFake) with > 100k lumi sections.
Maybe we could print lfn, run and # of lumis inside each run. What do you think?

File: src/python/WMCore/WorkQueue/WorkQueue.py (hunk 3 above)
Comment (amaltaro) [reply src/python/WMCore/WorkQueue/WorkQueue.py]: It should be enough to know the run and amount of lumis recovered for each of them.

File: src/python/WMCore/WMBS/File.py (hunk 2 above)
Comment (ticoann) [reply src/python/WMCore/WMBS/File.py]: So len(runLumiBinds) only prints number of lfns?

> Maybe we could print lfn, run and # of lumis inside each run. What do you think?

Yes I think that will give more information.

File: src/python/WMCore/WorkQueue/WorkQueue.py (hunk 3 above)
Comment (ticoann) [reply src/python/WMCore/WorkQueue/WorkQueue.py]: OK

**Comment by amaltaro** [comment]
Thanks for the review, Seangchan. I think we need to track 3 things:
1. that we pull all the lumi sections from ACDCServer (thus all lfns/runs/lumis)
2. that we don't miss any of this information until we get to File.addFilesToWMBSInBulk
3. this is a bit more complicated, but we should be able to see what are the binds we use for wmbs insertion. Another problem here is that we have one entry per lumi, and there is no way to know whether we're handling it for ACDC or standard workflows.

Do you see anything missing?

**Comment by ticoann** [comment]
Thanks Alan, That make sense. I was wondering since the missing lumi case examples has the same lfn entries with different lumis. we maybe should focus on that as well.  Printing out what lfn and has more than 1 entries in ACDC.

**Comment by amaltaro** [comment]
Since WorkQueueManager didn't crash in my testbed agent, I'm pushing this one to the 3 new production nodes.

**Comment by amaltaro** [comment]
This patch was already merged for 1.1.8.patch2

**Comment by ticoann** [comment]
@amaltaro, Alan do we still need to keep this around?

**Comment by amaltaro** [comment]
Absolutely not, it's from 2017 :)
